lifetimes/utils.py

```diff
@@ -259,6 +259,11 @@ def summary_data_from_transaction_data(transactions, customer_id_col, datetime_c
         customers['monetary_value'] = repeated_transactions.groupby(customer_id_col)[monetary_value_col].mean().fillna(0)
         summary_columns.append('monetary_value')
 
+    # fixes issue with pandas 0.24.* where columns below are offsets
+    if customers['T'].dtype != float:
+        customers['T'] = customers['T'].apply(lambda x: x.n)
```

Comment:
what does `x.n` do? I've never seen that syntax before

Reply:
from http://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.tseries.offsets.DateOffset.html `.n` is the attribute that represents the number of time periods in the offset

Reply:
hm, can you point to the changelog in pandas 0.24 that causes this?